### PR TITLE
Allow Dataset DOI in check permissions API request

### LINKED ISSUE #8994: API: Slash "/" is not allowed in path parameter when checking permissions
**What steps does it take to reproduce the issue?**
Calling API at "/api/admin/permissions/{dvo}" with global identifier as dvo fails. Slashes are not allowed as path parameters by default and escaping them with %2F does not work.
E.g. `http://locahost:8080/api/admin/permissions/doi:10.80442/5CPZVH` and  `http://locahost:8080/api/admin/permissions/doi:10.80442%2F5CPZVH` both fail

**What happens?**
Error: 404 not found.

**To whom does it occur (all users, curators, superusers)?**
Superusers (only superusers can check permissions with api)

**What did you expect to happen?**
A valid response with listed permissions for a given object.

**Which version of Dataverse are you using?**
Develop.

Allowing slashes in dvo path parameter solves the problem
`@Path("permissions/{dvo:.+}")`
`http://locahost:8080/api/admin/permissions/doi:10.80442/5CPZVH` works with this setting.

### PR DESCRIPTION
**Pull request by ErykKul**
**What this PR does / why we need it**:
It allows using a global identifier, e.g., doi, when checking permissions.

**Which issue(s) this PR closes**:

Closes #8994

### CODE REVIEW

**Comment by ErykKul** [comment]
The workaround here is to first get the dataset with the API, the response contains the technical (DB) ID that can be used (it does not contain slashes, it is an integer) to check the permissions.

**Review by qqmyers — CHANGES_REQUESTED** [review]
The fix looks like it works for the case where the dataset PID exists, but it will return 500 errors instead of 404  when it isn't found.

File: src/main/java/edu/harvard/iq/dataverse/api/Admin.java [hunk 1]
```diff
@@ -1334,7 +1335,10 @@ public Response findPermissonsOn(@Context ContainerRequestContext crc, @PathPara
 		try {
 			DvObject dvObj = findDvo(dvo);
 			if (dvObj == null) {
-				return notFound("DvObject " + dvo + " not found");
+				dvObj = findDatasetOrDie(dvo);
+				if (dvObj == null) {
```
[the remark this answers is not in the document]
Comment (ErykKul) [reply src/main/java/edu/harvard/iq/dataverse/api/Admin.java]: I have changed how findDvo works (it returns not null object now, or it throws WrappedResponse as thrown by the findDatasetOrDie). Then I cleaned it up a little bit. I think it is better now.

**Review by qqmyers — APPROVED** [review]

**Comment by sekmiller** [comment]
Hey @ErykKul, I'd like to pick this up but there's a merge conflict at the moment - so if you could resolve it I'll be able to test. Thanks!

**Comment by qqmyers** [comment]
Weird - the web editor to resolve the conflict shows 0 conflicts in that file...

**Comment by ErykKul** [comment]
@sekmiller @qqmyers I see that my PRs moved, but I am on vacation right now for another week, I will be back on 15th. I will re-merge the develop branch to them, do the necessary changes and retest once that I will be back at work. Some of them are open for quite some time, so they need some extra attention from me before merging, Thanks!

**Comment by ErykKul** [comment]
@sekmiller @qqmyers I had to manually merge the `AbstractApiBean.java`, but there where no conflicts to solve. Strange. It is now merged with the develop branch.

**Comment by sekmiller** [comment]
Hey @ErykKul This looks great. Thanks! The only thing I need you to do before I can merge it is for you to update the doc so that the url is in double quotes as in  curl -H "X-Dataverse-key:$API_TOKEN" "$SERVER_URL/api/admin/permissions/:persistentId?persistentId=$PERSISTENT_IDENTIFIER" - I would do it myself but I don't think I have permission.

**Comment by ErykKul** [comment]
@sekmiller Thanks for catching that! I have just added the missing double quotes.

**Pull request merged**